14Ianuarie/src/ro/sci/ale/ian14/store/MainStore.java

```diff
@@ -0,0 +1,237 @@
+                            break;
+                        case 2:
+                            addProduct("2");
+                            break;
+                        default:
+                            System.out.println("\nInvalid Choice");
+                    }
+
+                    break;
+
+                case 2:
+                    printList();
+                    sellProduct();
+
+                    break;
+
+                case 3:
+                    System.out.println(LocalDate.now());
+                    showSellListRepot();
+                    break;
+
+                case 4:
+                    System.out.println("Thanks ");
+                    quit = true;
+                    break;
+
+                default:
+                    System.out.println("\nInvalid Choice");
+            }
+        }
+        while (!quit);
+    }
+
+    /**
+     * method to input a new product with all the details and add it to the list of products
+     *
+     * @param productType - the type of product: animal or vegetable
+     */
+    private static void addProduct(String productType) {
+        String name;
+        int productID;
+        double price;
+        String validityDate;
+        double weight;
+        int quantity;
+        int storageTemp;
+        String vitaminsList;
+
+        System.out.println("Enter name: ");
+        name = scan.next();
+
+        System.out.println("Enter price in LEI: ");
+        price = scan.nextDouble();
+
+        System.out.println("Enter validityDate in format: YYYY-MM-DD ");
+        validityDate = scan.next();
+
+        System.out.println("Enter weight: ");
+        weight = scan.nextDouble();
+
+        System.out.println("Enter quantity: ");
+        quantity = scan.nextInt();
+
+        if (productType.equals("1")) {
+            System.out.println("Enter storage temperature: ");
+            storageTemp = scan.nextInt();
+
+            productID = animalList.size() + vegetableProductList.size();
+
+            AnimalProduct animal = new AnimalProduct(name, productID, price, validityDate,
+                    weight, quantity, storageTemp);
+            animalList.add(animal);
+
+            System.out.println(animal.getProductID());
+
+        } else {
+            System.out.println("Enter vitamin list: ");
+            vitaminsList = scan.nextLine();
+
+            productID = vegetableProductList.size() + animalList.size();
+            VegetableProduct vegetable = new VegetableProduct(name, productID, price, validityDate,
+                    weight, quantity, vitaminsList);
+            vegetableProductList.add(vegetable);
+        }
+    }
+
+    /**
+     * method to sell a product based on the id and quantity given by the user
+     *
+     */
+    public static void sellProduct() {
+        System.out.println("Please insert product Id: ");
+        String id = scan.next();
+
+        int quantity;
+        int i = 0;
+
+        for (Product x : animalList) {
```

Comment:
because you have animals and vegetables separately you need to do nearly 2 identical for loops here.